### PR TITLE
Errors

### PR DESCRIPTION
**Pull request by george-cosma**

### CODE REVIEW

File: tockloader-cli/src/main.rs [hunk 1]
```diff
@@ -17,140 +16,135 @@ use tockloader_lib::{
...
-    if let Err(e) = &result {
-        eprintln!("\n{}", e);
-    }
-    result
+async fn main() -> Result<()> {
+    run().await
 }
 
-async fn run() -> Result<(), TockloaderError> {
+async fn run() -> Result<()> {
```
Comment (george-cosma) [inline tockloader-cli/src/main.rs:23]: Main direct in loc "run"

File: tockloader-lib/src/tabs/tab.rs [hunk 2]
```diff
@@ -22,36 +22,48 @@ impl Tab {
...
-            let mut file = file.unwrap();
-            let path = file.path().unwrap();
-            let file_name = path.file_name().unwrap().to_str().unwrap().to_owned();
+        let file = File::open(path).map_err(TockloaderError::UnusableTab)?;
+        let mut archive = Archive::new(file);
+        for file in archive
+            .entries()
+            .map_err(TockloaderError::UnusableTab)?
+        {
+            match file {
```
Comment (george-cosma) [inline tockloader-lib/src/tabs/tab.rs:31]: .map_err(TockloaderError::UnusableTab)?

File: tockloader-lib/src/tabs/tab.rs [hunk 3]
```diff
@@ -22,36 +22,48 @@ impl Tab {
...
+                Ok(mut file) => {
+                    let path = file.path().map_err(TockloaderError::UnusableTab)?;
+                    match path.file_name() {
```
Comment (george-cosma) [inline tockloader-lib/src/tabs/tab.rs:34]: let file_name = match path.file_name().and_then(|name| name.to_str()) {
                        Some(name) => name,
                        None => continue,
                    };

File: tockloader-lib/src/tabs/tab.rs [hunk 4]
```diff
@@ -22,36 +22,48 @@ impl Tab {
...
+                        None => continue,
+                    }
+                }
+                Err(e) => return Err(TockloaderError::UnusableTab(e)),
             }
         }
 
-        if metadata.is_none() {
-            panic!("No metadata.toml found in tab");
+        match metadata {
```
Comment (george-cosma) [inline tockloader-lib/src/tabs/tab.rs:60]: map_err

**Comment by george-cosma** [comment]
+ cargo fmt :D

File: tockloader-lib/src/tabs/tab.rs (hunk 4 above)
Comment (micuami) [reply tockloader-lib/src/tabs/tab.rs:60]: metadata is an Option so I can't use map_err

Comment (george-cosma) [reply tockloader-lib/src/tabs/tab.rs:60]: ok_or

Comment (george-cosma) [reply tockloader-lib/src/tabs/tab.rs:60]: Actually, It's fine this way. I just notice you make it go from metadata to a TAB. It's better as-is actually

**Review by micuami — APPROVED** [review]

**Pull request merged**